concepts/0094-cross-domain-messaging/README.md

```diff
@@ -0,0 +1,302 @@
+
+1. **Sender Encapsulation**: We SHOULD minimize what the Receiver has to know about the domain (routing tree or agent infrastructure) of the Sender in order for them to communicate.
+2. **Receiver Encapsulation**: We SHOULD minimize what the Sender has to know about the domain (routing tree or agent infrastructure) of the Receiver in order for them to communicate.
+3. **Independent Keys**: Private signing keys SHOULD NOT be shared between agents; each agent SHOULD be separately identifiable for accounting and authorization/revocation purposes.
+4. ***Need To Know* Information Sharing**: Information made available to intermediary agents between the Sender and Receiver SHOULD be minimized to what is needed to perform the agent's role in the process.
+
+### Assumptions
+
+The following are assumptions upon which this RFC is predicated.
+
+#### Terminology
+
+The following terms are used in this RFC with the following meanings:
+
+- Domain - a set of Agents collaborating on behalf of an Identity
+  - It's assumed that the Agents of a Domain were developed by a single vendor and so MAY have implementation-specific mechanisms for tracking extra information about one another.
+  - An example of two Domains is provided in the image below.
+- Edge Agents - the Agents involved in sending (creating) and receiving (processing) an Agent Message
+  - Sender - the Agent sending an original Agent Message
+  - Receiver - the Agent intended to receive an Agent Message from the Sender
+  - **Note**: A message MAY pass through many Agents between the Sender and Receiver
+- Domain Endpoint - a physical endpoint for messages into domains that MUST be assumed to be shared across many Identities
+  - Shared by many identities (e.g. https://endpoint.agentsRus.com)
+  - Agency - the handler for messages sent to the Domain Endpoint.
+- Mediators - agents explicitly known to the sender as part of routing messages to a Receiver.
+- DID - reference to the literal Decentralized ID text
+  - e.g. did:sov:1234abcd
+- DID#keyname - reference to the DID appended with "#" and a specific key from the DIDDoc
+  - e.g. did:sov:1234abcd#1 references key "1" of the "did:sov:1234abcd" DIDDoc.
+  - **Note**: The #keyname is NOT the actual Public Key - it's a reference to an entry in the DIDDoc that contains the Public Key.
+
+#### DIDDoc
+
+The term "DIDDoc" is used in this RFC as it is defined in the [DID Specification](https://w3c-ccg.github.io/did-spec/):
+
+- a collection of public keys and endpoints,
+- controlled by an identity,
+- associated with a DID, and
+- used for a relationship.
+
+A DID can be resolved to get its corresponding DIDDoc by any Agent that needs access to the DIDDoc. This is true whether talking about a DID on a Public Ledger, or a pairwise DID persisted only to the parties of the relationship. In the case of pairwise DIDs, it's the (implementation specific) domain's responsibility to ensure such resolution is available to all Agents requiring it.
```

Comment:
@swcurran aren't we specifically referring to peer DIDs here? Why not link to that spec?

Reply:
Yes - the link was later in the doc. I've moved up the link to the peer:did method to the first reference and made sure that each reference to pairwise DID also references the peer:did method.